File: Java/src/main/java/TicketOffice.java
import java.util.List;

public class TicketOffice {
    
    private TrainDataService trainDataService;

    public TicketOffice(String trainDataService, String bookingReferenceService) {
		//TODO: implement this code!
    }

    public TicketOffice(TrainDataService trainDataService, Object bookingReferenceService) {
        this.trainDataService = trainDataService;
    }

    public Reservation makeReservation(ReservationRequest request) {
        List<Seat> trainInformation = trainDataService.getTrainInformation("express_2000");
		return new Reservation("", trainInformation, "");
    }

}

File: csharp/TicketOfficeService/TicketOffice.cs
namespace TicketOfficeService;

public record ReservationRequest(string TrainId, int SeatCount);
public record Reservation(string TrainId, string BookingId, List<Seat> Seats);
public record Seat(string Coach, int SeatNumber);

public class TicketOffice
{
    public Reservation MakeReservation(ReservationRequest request)
    {
        //TODO: implement this code!
        throw new NotImplementedException();
    }
}